### PR TITLE
UCP/PROTO: Enable test_tag_xfer on new protocol and fix related issues

### PR DESCRIPTION
**Pull request by yosefe**
## Why
- Enable rndv on non-contig data
- Enable more tests for new protocols and fix related issues

### CODE REVIEW

**Comment by yosefe** [comment]
@evgeny-leksikov @hoopoepg can you pls take a look?

File: src/ucp/proto/proto_single.inl [hunk 1]
```diff
@@ -44,21 +44,25 @@ ucp_proto_am_bcopy_single_send(ucp_request_t *req, ucp_am_id_t am_id,
...
         req->send.lane = lane;
         return UCS_ERR_NO_RESOURCE;
-    } else if (status != UCS_INPROGRESS) {
+    } else {
         ucp_proto_request_abort(req, status);
+        return UCS_OK;
+    }
+
+    if (complete_func != NULL) {
+        complete_func(req);
```
Comment (hoopoepg) [inline src/ucp/proto/proto_single.inl:65]: ignore return value?
may be ```return complete_func(req)``` ?

File: src/ucp/proto/proto_multi.inl [hunk 2]
```diff
@@ -48,16 +48,23 @@ ucp_proto_multi_max_payload(ucp_request_t *req,
...
+    max_payload = ucs_min(ucp_proto_multi_scaled_length(lpriv, length),
+                          max_frag);
     ucs_assertv(max_payload > 0,
-                "length=%zu weight=%.2f scaled_length=%zu max_frag=%zu "
-                "hdr_size=%zu",
+                "length=%zu weight=%zu%% lpriv->max_frag=%zu hdr_size=%zu",
                 req->send.state.dt_iter.length,
-                lpriv->weight / (double)UCS_BIT(UCP_PROTO_MULTI_WEIGHT_SHIFT),
-                scaled_length, lpriv->max_frag, hdr_size);
+                ucp_proto_multi_scaled_length(lpriv, 100), lpriv->max_frag,
```
Comment (hoopoepg) [inline src/ucp/proto/proto_multi.inl:66]: why 100?

**Review by hoopoepg — APPROVED** [review]
minor comments

File: src/ucp/proto/proto_multi.inl (hunk 2 above)
Comment (yosefe) [reply src/ucp/proto/proto_multi.inl:66]: this prints percentage of data on this lane so it's 100*ratio

**Comment by yosefe** [comment]
bot:pipe:retest

**Comment by yosefe** [comment]
@evgeny-leksikov fixed

**Review by evgeny-leksikov — APPROVED** [review]

**Pull request merged**